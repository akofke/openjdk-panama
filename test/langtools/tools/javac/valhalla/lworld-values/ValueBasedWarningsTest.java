/*
 * @test /nodynamiccopyright/
 * @summary Check that javac emits warnings rather than errors for null violation with value based classes.
 * @compile/fail/ref=ValueBasedWarningsTest.out -Werror -XDallowValueBasedClasses -XDrawDiagnostics -XDdev ValueBasedWarningsTest.java
 * @compile/fail/ref=ValueBasedWarningsTest2.out -Werror -XDrawDiagnostics -XDdev ValueBasedWarningsTest.java
 */

public class ValueBasedWarningsTest {
    @ValueBased
    final __ByValue class X {
        final int x = 10;
        void foo(X x1, X x2) {
            x1 = null;
            x2 = (X) null;
            if (null instanceof X) {}
        }
    }
}
