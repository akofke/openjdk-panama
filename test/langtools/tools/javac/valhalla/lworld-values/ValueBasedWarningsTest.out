ValueBasedWarningsTest.java:13:18: compiler.warn.suspicious.mix.of.null.with.value.based.class: ValueBasedWarningsTest.X
ValueBasedWarningsTest.java:14:22: compiler.warn.suspicious.mix.of.null.with.value.based.class: ValueBasedWarningsTest.X
ValueBasedWarningsTest.java:15:17: compiler.warn.suspicious.mix.of.null.with.value.based.class: ValueBasedWarningsTest.X
- compiler.err.warnings.and.werror
1 error
3 warnings
